pubsub/api/PubsubTest.cs

```diff
@@ -0,0 +1,410 @@
+            // [END list_topics]
+            return topics;
+        }
+
+        public IEnumerable<Subscription> ListSubscriptions(SubscriberClient 
+            subscriber)
+        {
+            // [START list_subscriptions]
+            string projectName = PublisherClient.FormatProjectName(_projectId);
+            IEnumerable<Subscription> subscriptions = 
+                subscriber.ListSubscriptions(projectName);
+            // [END list_subscriptions]
+            return subscriptions;
+        }
+
+        public void DeleteSubscription(string subscriptionId, SubscriberClient 
+            subscriber)
+        {
+            // [START delete_subscription]
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, 
+                subscriptionId);
+            subscriber.DeleteSubscription(subscriptionName);
+            // [END delete_subscription]
+        }
+
+        public void DeleteTopic(string topicId, PublisherClient publisher)
+        {
+            // [START delete_topic]
+            string topicName = PublisherClient.FormatTopicName(_projectId, 
+                topicId);
+            publisher.DeleteTopic(topicName);
+            // [END delete_topic]
+        }
+
+        // [START retry]
+        /// <summary>
+        /// Retry the action when a Grpc.Core.RpcException is thrown.
+        /// </summary>
+        private T RetryRpc<T>(Func<T> action)
+        {
+            List<Grpc.Core.RpcException> exceptions = null;
+            var delayMs = _retryDelayMs;
+            for (int tryCount = 0; tryCount < _retryCount; ++tryCount)
+            {
+                try
+                {
+                    return action();
+                }
+                catch (Grpc.Core.RpcException e)
+                {
+                    if (exceptions == null)
+                        exceptions = new List<Grpc.Core.RpcException>();
+                    exceptions.Add(e);
+                }
+                System.Threading.Thread.Sleep(delayMs);
+                delayMs *= 2;  // Exponential back-off.
+            }
+            throw new AggregateException(exceptions);
+        }
+
+        private void RetryRpc(Action action)
+        {
+            RetryRpc(() => { action(); return 0; });
+        }
+
+        public int RpcRetry(string topicId, string subscriptionId,
+            PublisherClient publisher, SubscriberClient subscriber)
+        {
+            int tryCount = 0;
+            string topicName = PublisherClient.FormatTopicName(_projectId, topicId);
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, subscriptionId);
+            RetryRpc(() =>
+            {
+                //Create Topic only on second try
+                if (tryCount++ == 1)
+                {
+                    //Create Topic
+                    publisher.CreateTopic(topicName);

```

Comment:
I think I'd rather see the two API calls in two separate calls to RetryRpc().  That should also obviate tryCount.